packages/config/src/config.js

```diff
@@ -13,22 +54,26 @@ module.exports = ({
     useFileHash = true,
     betaEnv = 'ci',
     sassPrefix,
-    deployment
+    isProd,
+    isStandalone,
+    standalonePort,
+    proxyConfig
 } = {}) => {
     const filenameMask = `js/[name]${useFileHash ? '.[chunkhash]' : ''}.js`;
     return {
-        mode: mode || (process.env.NODE_ENV === 'production' ? 'production' : 'development'),
-        devtool: 'source-map',
+        mode: mode || (isProd ? 'production' : 'development'),
+        devtool: isProd ? 'source-map' : 'cheap-module-source-map',
         optimization: {
-            minimize: (process.env.NODE_ENV || mode) === 'production',
+            minimize: isProd || mode === 'production',
+            runtimeChunk: true,
```

Comment:
Required to hot module reloading Federated Modules, might as well add now. Also nice for caching.